mantidproject/mantid
# Improve speed of table workspace display for large datasets

**Description of work.**
This PR improves the performance of table workspace display for large datasets. It does this by creating a custom table model, which loads data in batches. It should lead to speed improvements opening large tables and also improve the performance when performing column operations on the table, e.g. sorting. 

**To test:**
Load a table workspace and check everything functions as you'd expect
Load a big table workspace to observe the speed improvements, e.g. with the following script:

```
ws = Load("TOPAZ_3132_event.nxs")
tbl = CreateDetectorTable(ws)
```

Fixes #29636. 





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
